climada/hazard/centroids/centr.py

```diff
@@ -1244,152 +813,103 @@ def _from_hdf5(cls, data):
                     meta[key] = value[0]
                 else:
                     meta[key] = rasterio.Affine(*value)
-            centr = cls(meta=meta)
+            latitude, longitude = _meta_to_lat_lon(meta)
 
+        extra_values = {}
         for centr_name in data.keys():
-            if centr_name not in ('crs', 'lat', 'lon', 'meta'):
-                setattr(centr, centr_name, np.array(data.get(centr_name)))
-        return centr
-
-    @property
-    def crs(self):
-        """Get CRS of raster or vector."""
-        if self.meta:
-            return self.meta['crs']
-        if self.geometry.crs:
-            return self.geometry.crs
-        return DEF_CRS
+            if centr_name not in ('crs', 'lat', 'lon', 'meta', 'latitude', 'longitude'):
+                values = np.array(data.get(centr_name))
+                if latitude.size != 0 and values.size != 0 :
+                    extra_values[centr_name] = values
+
+        return gpd.GeoDataFrame(
+            extra_values,
+            geometry=gpd.points_from_xy(x=longitude, y=latitude, crs=crs)
+        )
 
-    @property
-    def size(self):
-        """Get number of pixels or points."""
-        if self.meta:
-            return int(self.meta['height'] * self.meta['width'])
-        return self.lat.size
+    '''
+    Private methods
+    '''
 
-    @property
-    def shape(self):
-        """Get shape of rastered data."""
-        try:
-            if self.meta:
-                return (self.meta['height'], self.meta['width'])
-            return (np.unique(self.lat).size, np.unique(self.lon).size)
-        except AttributeError:
-            return ()
 
-    @property
-    def total_bounds(self):
-        """Get total bounds (left, bottom, right, top)."""
-        if self.meta:
-            left = self.meta['transform'].xoff
-            right = left + self.meta['transform'][0] * self.meta['width']
-            if left > right:
-                left, right = right, left
-            top = self.meta['transform'].yoff
-            bottom = top + self.meta['transform'][4] * self.meta['height']
-            if bottom > top:
-                bottom, top = top, bottom
-            return left, bottom, right, top
-        return self.lon.min(), self.lat.min(), self.lon.max(), self.lat.max()
+    def _ne_crs_geom(self):
+        """Return `geometry` attribute in the CRS of Natural Earth.
```

Comment:
Is it necessary to have this method? Why not simply call `Centroids.to_crs`?

Reply:
For legacy it is simpler yes.